### ISSUE TITLE
crate indexing

### ISSUE THREAD

**Issue by durka** [SDBlumire/rust-lab#4]
It would be really cool if the site had an overall list of crates that users are working on (maybe you have to opt-in to making them public), with generated docs and everything. Whenever cargo gets multi-repo support, we could have a mini crate host.

**Comment by tilpner** [comment]
Doesn't it already support alternate registries via [`registry.index`](http://doc.crates.io/config.html#configuration-keys)?

**Comment by durka** [comment]
Yes, alternates, but not multiple AFAIK. That is, you can't say "try this
mirror first, but if a crate is not found there go to crates.io".

**Comment by tilpner** [comment]
Ah, right. It might or not be feasible/terrible to proxy crates.io, but that's probably not worth it.

I mean... it's easy to get the changes over time via rust-lang/crates.io-index, and then you can pull those crates directly from crates.io and cache them locally. Put that in a 1-minute cron-job or hook it up to some push-API and you're good. Though it's much cleaner to add multi-repo support to cargo instead, I agree.

**Comment by SDBlumire** [comment]
Agreed, I think leaving it dependant on pure crates.io wouldn't be a bad thing, but I agree with some kind of opts in list of projects being worked on, maybe being able to let people see you edit even if they can't contribute themselves? Like a code browser with live updates.
